app/src/main/java/unipd/se18/ocrcamera/MainApplication.java

```diff
@@ -0,0 +1,96 @@
+package com.example.mattia.fotocamera;
+
+import android.app.Application;
+import android.content.res.AssetManager;
+import android.util.Log;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public class MainApplication extends Application {
+
+    public static MainApplication instance = null;
```

Comment:
The access to this instance can be private, anyway you can just remove it and call methods like getExternalFilesDir() and getAssets() directly (line 30, 40 and 54) as they belong to Context class. 
If you need to get the external files dir from outside the application (line 16 - OcrManager) you may consider passing as argument the context and use it to call getExternalFilesDir() or creating an appropriate static string variable to embrace OOP encapsulation.